### PR TITLE
Migrate azure-security-attestation to Test Proxy

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Migrates `azure-security-attestation` to Test Proxy.

Updates creation of test resources to output the signing certificates and keys to file instead of requiring them to be updated manually in source code. When rerunning test records with a new resource these existing files simply need to be replaced by what is generated by the resource deployment script.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java [hunk 1]
```diff
@@ -60,16 +71,44 @@
...
+
+        if (interceptorManager.isPlaybackMode()) {
+            interceptorManager.addMatchers(Collections.singletonList(new CustomMatcher()
+                .setHeadersKeyOnlyMatch(Collections.singletonList("Authorization"))));
+        }
+    }
+
+    private static final TestMode TEST_MODE = getTestModeLocal();
+
+    static TestMode getTestModeLocal() {
```
Comment (samvaity) [inline sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java:96]: why do we need this?

File: sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java [hunk 2]
```diff
@@ -135,209 +175,181 @@ ClientTypes classifyClient(String clientUri) {
...
-            + "xwAl/C/m37z+qSOtOPbOv7Vwm7neeh5pwAFWS5TSTP5eYoh3Jy58C8aC7DdQEXFym14rFvn0EVeFLjbQ8GX+yHElBFwKBgQDGN4TrTGq"
-            + "r2jzrSEk2MuMdol6TOFi2h7avrc5WNQ7A0CTSKS6bOB2iQRZ5NGNXbO/tjc1O25AymRvtyot1TjI0POOuUqANg+G++8NlwV+jRLt4/bh"
-            + "vxq6V5poYz4vVqOt8XMoud6P2ZUEdyMbm8Mb5CyhgIgq40brP4svVrwFGwwKBgQDbZ7M9U38nBtHFIkJyRzE2iTp7P5pQjaNzPB9EiA9"
-            + "qSUA/ek3650fGk4bkvnFBtHo/00yBei9CrWlI5XrH5hWigRRp4SiI0PYCXRf3Y4iHw4rifmCdfO9pDWbtWxvTIPs7tyqw7ojNDmQTdxw"
-            + "bIV2mzMJy/ulxh2kLCK7jUt/RCQKBgF2GEJcc5S/Utu2km7aQ6AwxckIwUds9yXq2nSabvixHiTgH6k+eeIz3/OyPRVgYtFP3lilg+3a"
-            + "EO2NFL+vXS2TOXfcAUThPQ5JjI9y9EqzsXzBFNmK28eUUexy9wJK9Mu7osRSfqqM1PzYYZ5DLhnY3R4OkTD7L5kL0FYzrF8UVAoGBAKy"
-            + "hZZuD4sMS3WsDg9d+LfhnGcKTsBfovi7KSqHLLZmqJ6sxKXGoWe9IlGadfWk3VNIW/LQ+/hipZv4bsOCBcOc097+Ev23u6bl2Xm0LHq9"
-            + "FFUDqTfD8qTpQF+GWOEXfdV5NfwmwRc1MLcAa3BFwvRgG+yCoMZgP4PGE3rhdakWF";
+    static String getIsolatedSigningKeyBase64() {
+        return TEST_MODE == TestMode.PLAYBACK
```
Comment (samvaity) [inline sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java:288]: why aren't we using `interceptorManager` directly?

**Review by samvaity — APPROVED** [review]
Other changes lgtm!

File: sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java:96]: There's nowhere in azure-core-test that exposes the static TestMode value. We should think about exposing that publicly next release.

File: sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java:288]: These are static methods that get used by ParameterizedTests, so they can't take method arguments.

**Comment by alzimmermsft** [comment]
Overriding check enforcer as the failing linting checks are unrelated to this change.

**Pull request merged**